src/utils.ts

```diff
@@ -38,6 +39,37 @@ const scoreApiHeaders = {
   'Content-Type': 'application/json'
 };
 
+const DEFAULT_SCORE_API_URL = 'https://score.snapshot.org';
+
+const formatScoreAPIUrl = (url = DEFAULT_SCORE_API_URL) => {
+  const scoreURL = new URL(url);
+  const apiKey = scoreURL.searchParams.get('apiKey');
+  let headers: any = { ...scoreApiHeaders };
+  if (apiKey) {
+    scoreURL.searchParams.delete('apiKey');
+    headers = { ...scoreApiHeaders, 'X-API-KEY': apiKey };
+  }
+  return {
+    url: scoreURL.toString(),
+    headers
+  };
+};
+
+const parseScoreAPIResponse = async (res: any) => {
+  let json: any = await res.text();
+  try {
+    json = JSON.parse(json);
+  } catch (e) {
+    return Promise.reject({
+      code: 500,
```

Comment:
We could get the true status code from `res`, and only fallback to 500 if missing